File: clients/algoliasearch-client-kotlin/client/src/commonMain/kotlin/com/algolia/client/api/ApiClient.kt
package com.algolia.client.api

import com.algolia.client.configuration.ClientOptions
import com.algolia.client.transport.Requester

/**
 * An interface representing an API client with specific properties and options.
 *
 * @property appId The unique identifier for the application using the API client.
 * @property apiKey The API key used for authentication.
 * @property options A set of custom client options to configure the behavior of the API client.
 */
public sealed interface ApiClient {
  public val appId: String
  public val apiKey: String
  public val options: ClientOptions
  public val requester: Requester
}

File: clients/algoliasearch-client-dart/packages/client_core/lib/src/api_client.dart
import 'package:algolia_client_core/src/config/client_options.dart';

/// An abstract class representing an API client with specific properties and options.
abstract interface class ApiClient {
  /// The unique identifier for the application using the API client.
  String get appId;

  /// The API key used for authentication.
  String get apiKey;

  /// A set of custom client options to configure the behavior of the API client.
  ClientOptions get options;

  /// Dispose of underlying resources.
  void dispose();
}
